v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/transforms/AvroTransform.java

```diff
@@ -61,15 +66,27 @@
  */
 public class AvroTransform
     extends PTransform<PCollection<KafkaRecord<byte[], byte[]>>, WriteResult> {
-
-  private static final Logger LOG = LoggerFactory.getLogger(AvroTransform.class);
-
+  private BadRecordRouter badRecordRouter = BadRecordRouter.RECORDING_ROUTER;
+  private List<ErrorHandler<BadRecord, ?>> errorHandlers;
+  private static final TupleTag<FailsafeElement<KafkaRecord<byte[], byte[]>, GenericRecord>>
+      SUCCESS_GENERIC_RECORDS = new TupleTag<>();
+  private static final TupleTag<FailsafeElement<KafkaRecord<byte[], byte[]>, TableRow>>
+      SUCCESS_TABLE_ROW = new TupleTag<>();
+  private static final KafkaRecordCoder kafkaRecordCoder =
+      KafkaRecordCoder.of(
+          NullableCoder.of(ByteArrayCoder.of()), NullableCoder.of(ByteArrayCoder.of()));
   private KafkaToBigQueryFlexOptions options;
```

Comment:
In general, transforms shouldn't have a dependency on pipeline options.

1) On high level, from the design perspective, transforms are supposed to be "building blocks", reusable across multiple pipelines. Making them dependent on a specific Options class makes them unusable with other Options classes and other pipelines (or you'll end up with some overly complicated logic that tries to construct the transform from many different Options classes).
2) On low level, it's also better to keep the amount of class variable to a minimum because Transforms are serialized, i.e. it's better to keep only those variables that the transform actually needs instead of the whole Options class because it contains many other options not used in the transform. This is to reduce the size of the serialized transform. Not very important as its size shouldn't be huge anyway, but storing the whole Options class is excessive.

Typical pattern is to read Options int the pipeline class and pass the option value to the transform, example: https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/4b06b8b244043565c9aa4b689e10a4be887c400a/v2/dataplex/src/main/java/com/google/cloud/teleport/v2/templates/DataplexBigQueryToGcs.java#L328